src/main/java/prefs/ConfigFileHandler.java

```diff
@@ -12,67 +12,131 @@
 import java.time.LocalDateTime;
 import java.time.ZoneId;
 
+/**
+ * A general handler for HubTurbo config files. Serialises and deserialises config files.
+ *
+ * To retrieve specific config data that is stored in a file:
+ * // example 1 - GlobalConfig type
+ * ConfigFileHandler cfhGlobal = new ConfigFileHandler("config/dir", "globalconfigfile.name", new GlobalConfig());
+ * GlobalConfig gc = cfhGlobal.getConfig();
+ *
+ * // example 2 - UserConfig type
+ * ConfigFileHandler cfhUser = new ConfigFileHandler("config/dir", "userconfigfile.name", new UserConfig());
+ * UserConfig gc = cfhUser.getConfig();
+ *
+ * To save a specified Config object to a file:
+ * ConfigFileHandler cfhGlobal = new ConfigFileHandler("config/dir", "globalconfigfile.name", new GlobalConfig());
+ * cfhGlobal.saveConfig();
+ */
 public class ConfigFileHandler {
 
     private static final Logger logger = LogManager.getLogger(ConfigFileHandler.class.getName());
 
     private static final String CHARSET = "UTF-8";
-    private final String globalConfigFileName;
+    private final String configFileName;
     private final String configDirectory;
 
+    private Config config;
+
     private Gson gson;
 
-    public ConfigFileHandler(String configDirectory, String globalConfigFileName) {
-        this.globalConfigFileName = globalConfigFileName;
+    /**
+     * Constructs a new Config File Handler that handles a specific Config type.
+     * @param configDirectory The directory that the config file is held in
+     * @param configFileName The file name of the config file
+     * @param config The config file object to deserialise into.
+     */
+    public ConfigFileHandler(String configDirectory, String configFileName, Config config) {
+        this.configFileName = configFileName;
         this.configDirectory = configDirectory;
+        this.config = config;
 
         setupGson();
         ensureDirectoryExists();
     }
 
-    private void ensureDirectoryExists() {
+    /**
+     * Ensures that the directory specified in the attribute exists.
+     * Attempts to create the directory if it does not exist.
+     * @return True if the directory exists or was created.
+     *         False if by the end of the method, the directory still doesn't exist
+     */
+    private boolean ensureDirectoryExists() {
         File directory = new File(configDirectory);
-        boolean directoryNonExistent = !directory.exists() || !directory.isDirectory();
-        boolean exists = true;
+        boolean exists = directory.exists();
```

Comment:
I prefer the original variables here since it makes much more sense to check both `file.exists` and `file.isDirectory` so that the `exists` variable does not disambiguate to mean the directory already exists. Maybe a more concise implementation would be something like:

```
boolean directoryExists = directory.exists() && directory.isDirectory();
if (!directoryExists) {
  if (directory.mkdirs()) {
    return true;
  } else {
    logger.warn("Could not create config file directory");
    return false;
}
```